src/components/custom-field-input-number/custom-field-input-number.jsx

```diff
@@ -0,0 +1,72 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef, useState } from 'react';
+
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
+import styles from '../custom-field-input-text/custom-field-input-text.css';
+
+function getRootClassName(className, error, disabled) {
+  if (disabled) {
+    return `${className} ${styles.disabled}`;
+  }
+
+  if (error) {
+    return `${className} ${styles.error}`;
+  }
+
+  return className;
+}
+
+export default function CustomFieldInputNumber(props) {
+  const inputRef = useRef(null);
+  const [invalid, setInvalid] = useState(false);
+
+  function handleOnKeyUp(event) {
+    setInvalid(!event.target.checkValidity());
+  }
+
+  useEffect(() => {
+    if (!inputRef.current) return;
+
+    setInvalid(!inputRef.current.validity.valid);
+  });
+
+  return (
+    <CustomFieldInputText
+      className={getRootClassName(props.className, invalid, props.disabled)}
+      disabled={props.disabled}
+      error={invalid}
+      id={props.id}
+      inputRef={inputRef}
+      label={props.label}
+      max={2 ** 31} // This value is sourced from the API
```

Comment:
Could we make these named constants at the top of the file, with names that imply their source/purpose?

Reply:
Done!